### DISCUSSION TITLE
Debug Tips

### DISCUSSION THREAD

**Discussion by mmann1123** [jgrss/geowombat#192] [Q&A]
@jgrss 
My debugging routine has been to have a env with all the depedencies installed but not geowombat, and inserting a debug.py script in the geowombat foolder and importing gw locally. 

Since geowombat has moved into src/geowombat this has gone tits up. Even if I put the script in the src folder, its not importing geowombat. I need to import src.geowombat and update all references in the code.  What's your debugging routine?

**Comment by jgrss** [comment]
I'm not sure I fully follow what you mean by 'debugging routine'. Do mean just testing code?

1. I isolate a new virtual environment.

```commandline
pyenv virtualenv 3.8.12 test.venv
pyenv activate test.venv
```

2. Create a branch and install an editable version of the branch

```python
cd geowombat/
git checkout -b jgrss/sandbox
(test.venv) pip install --user -e .
```

3. Test scripts, edit the geowombat branch, and changes are automatically updated (i.e., you don't have to reinstall)

```
cd script_location/
(test.venv) python script.py
```

Is that what you are asking?

**Reply by mmann1123** [reply]
Not exactly, my issue is being able to run the debugger. At least in vscode I've found it difficult to get the debugger to work if the module is installed like `pip install -e .`. 

So I have always avoided installing and just importing geowombat locally but not pip installing it.  

Maybe it would be worth knowing what IDE you use, and how you have it set up to run a debugger if you have geowombat pip installed.

**Reply by jgrss** [reply]
Oh, I see. I use VSCode but don't really like using IDE debuggers. I tend to just put a trace in the script and run it. E.g.,

```python
import ipdb; ipdb.set_trace()
```

**Comment by mmann1123** [comment]
No rush on this, I just figured out the issue. I forgot to append the path before the import:
```
import sys 
sys.path.append("/") 
sys.path.append("/geowombat") 
```
Silly
